sympy/solvers/solveset.py

```diff
@@ -339,10 +339,21 @@ def _invert_complex(f, g_ys, symbol):
                                imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, exp):
-        if isinstance(g_ys, FiniteSet):
+        from sympy.abc import k,n
+        if isinstance(g_ys,list) and not isinstance(g_ys, FiniteSet) \
+                        and not isinstance(g_ys,ImageSet):
```

Comment:
If `g_ys` is a `list` then it won't be an instance of either FiniteSet or ImageSet

Reply:
Nice point! 
I'll rectify it. Sorry for the inconvenience!